File: Musixmatch_lyrics_API_App/app/src/main/java/alexejantonov/com/musixmatch_lyrics_api_app/ui/artists_and_tracks/ArtistsAndTracksListView.java
package alexejantonov.com.musixmatch_lyrics_api_app.ui.artists_and_tracks;

import com.arellomobile.mvp.MvpView;
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy;
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;

import java.util.List;

import alexejantonov.com.musixmatch_lyrics_api_app.ui.Base.BaseData;

@StateStrategyType(AddToEndSingleStrategy.class)
interface ArtistsAndTracksListView extends MvpView {
	void showData(List<BaseData> data);
}

File: Musixmatch_lyrics_API_App/app/src/main/java/alexejantonov/com/musixmatch_lyrics_api_app/ui/artists_and_tracks/ArtistsAndTracksListView.kt
package alexejantonov.com.musixmatch_lyrics_api_app.ui.artists_and_tracks

import alexejantonov.com.musixmatch_lyrics_api_app.ui.Base.BaseData
import com.arellomobile.mvp.MvpView
import com.arellomobile.mvp.viewstate.strategy.AddToEndSingleStrategy
import com.arellomobile.mvp.viewstate.strategy.StateStrategyType

@StateStrategyType(AddToEndSingleStrategy::class)
interface ArtistsAndTracksListView : MvpView {
  fun showData(data: List<BaseData>)
  fun showLoading()
  fun hideLoading()
}
